### PR TITLE
http/client: Fix content length body overflow check (and a bit more)

### PR DESCRIPTION
**Pull request by xemul**
When writing a data over specified content length, the helper output_stream throws. However, there's a corner case when it doesn't -- when the body is byte-to-byte full and extra bytes are put, they are silently ignored.

This PR fixes it and adds a unit test for that corner case. Also the test covers zero-copy behavior.

### CODE REVIEW

File: src/http/common.cc [hunk 1]
```diff
@@ -173,11 +173,11 @@ class http_content_length_data_sink_impl : public data_sink_impl {
...
     virtual future<> put(temporary_buffer<char> buf) override {
-        if (buf.size() == 0 || _bytes_written == _limit) {
+        auto size = buf.size();
+        if (size == 0) {
             return make_ready_future<>();
         }
 
-        auto size = buf.size();
         if (_bytes_written + size > _limit) {
             return make_exception_future<>(std::runtime_error(format("body conent length overflow: want {} limit {}", _bytes_written + buf.size(), _limit)));
```
Comment (avikivity) [inline src/http/common.cc]: test?

**Comment by xemul** [comment]
upd:
- fixed a misprint in exception message
- added unit test

**Comment by xemul** [comment]
@avikivity , please re-review
